exec/java-exec/src/main/codegen/templates/Decimal/CastIntDecimal.java

```diff
@@ -64,7 +64,15 @@ public void setup() {
 
     public void eval() {
         out.scale = (int) scale.value;
-        out.precision = (int) precision.value;
```

Comment:
I am not too familiar with the decimal type but taking a closer look at the fix I don't think we should ignore the supplied precision.  Your fix will compute the precision in all cases, which would be a performance overhead.  It should be computing only when supplied precision is 0.  Thoughts ?